lib/modules/manager/helmv3/artifacts.spec.ts

```diff
@@ -723,6 +746,246 @@ describe('modules/manager/helmv3/artifacts', () => {
     expect(execSnapshots).toMatchSnapshot();
   });
 
+  it('supports ECR authentication', async () => {
+    mockEcrAuthResolve({
+      authorizationData: [
+        { authorizationToken: toBase64('token-username:token-password') },
+      ],
+    });
+
+    hostRules.add({
+      username: 'some-username',
+      password: 'some-password',
+      token: 'some-session-token',
+      hostType: 'docker',
+      matchHost: '123456789.dkr.ecr.us-east-1.amazonaws.com',
+    });
+
+    fs.getSiblingFileName.mockReturnValueOnce('Chart.lock');
+    fs.readLocalFile.mockResolvedValueOnce(ociLockFile1ECR as never);
+    const execSnapshots = mockExecAll();
+    fs.readLocalFile.mockResolvedValueOnce(ociLockFile2ECR as never);
+    fs.privateCacheDir.mockReturnValue(
+      '/tmp/renovate/cache/__renovate-private-cache'
+    );
+    fs.getParentDir.mockReturnValue('');
+
+    expect(
+      await helmv3.updateArtifacts({
+        packageFileName: 'Chart.yaml',
+        updatedDeps: [],
+        newPackageFileContent: chartFileECR,
+        config: {
+          ...config,
+          updateType: 'lockFileMaintenance',
+          registryAliases: {},
+        },
+      })
+    ).toMatchObject([
+      {
+        file: {
+          type: 'addition',
+          path: 'Chart.lock',
+          contents: ociLockFile2ECR,
+        },
+      },
+    ]);
+
+    const ecr = ecrMock.call(0).thisValue as ECRClient;
+    expect(await ecr.config.region()).toBe('us-east-1');
+    expect(await ecr.config.credentials()).toEqual({
+      accessKeyId: 'some-username',
+      secretAccessKey: 'some-password',
+      sessionToken: 'some-session-token',
+    });
+
+    expect(execSnapshots).toMatchObject([
+      {
+        cmd: 'helm registry login --username token-username --password token-password 123456789.dkr.ecr.us-east-1.amazonaws.com',
+      },
+      {
+        cmd: "helm dependency update ''",
+      },
+    ]);
```

Comment:
Hopefully this is what you were going for with `toMatchObject()` - I just tested the `cmd` properties, which I think makes the tests clearer.